client/packages.go

```diff
@@ -67,8 +67,30 @@ type Snap struct {
 }
 
 type AppInfo struct {
-	Name   string `json:"name"`
-	Daemon string `json:"daemon"`
+	Name         string `json:"name"`
+	DesktopFile  string `json:"desktop-file,omitempty"`
+	*ServiceInfo `json:",omitempty"`
+}
+
+func (a *AppInfo) IsService() bool {
+	if a == nil {
```

Comment:
not sure how common it is. In general if it's a public thing and it doesn't have an exporter, I like to handle the `nil` case gracefully where it makes sense